src/Features/CSharp/Portable/MakeRefStruct/CSharpMakeRefStructCodeFixProvider.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeRefStruct
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class CSharpMakeRefStructCodeFixProvider : CodeFixProvider
+    {
+        // Error CS8345: Field or auto-implemented property cannot be of certain type unless it is an instance member of a ref struct.
+        private const string CS8345 = nameof(CS8345);
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(CS8345);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var span = context.Span;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var structDeclaration = FindContainingStruct(root, span);
```

Comment:
Does the compiler error only occur inside a struct? What if we're inside a class? Even if it never occurs inside classes, it might be a good idea to verify that struct is not null.